kinder/pkg/cluster/manager/actions/kubeadm-upgrade.go

```diff
@@ -120,7 +120,7 @@ func upgradeKubeadmBinary(n *status.Node, upgradeVersion *K8sVersion.Version) er
 	return nil
 }
 
-func kubeadmUpgradeApply(c *status.Cluster, cp1 *status.Node, upgradeVersion *K8sVersion.Version, patchesDir string, featureGates string, wait time.Duration, vLevel int) error {
+func kubeadmUpgradeApply(c *status.Cluster, cp1 *status.Node, upgradeVersion *K8sVersion.Version, patchesDir string, featureGate string, wait time.Duration, vLevel int) error {
```

Comment:
Overall LGTM.

why change featureGates to featureGate? It is confusing.

Reply:
We can only support a single kubeadm feature-gate now.

Reply:
// FeatureGate option sets **a single kubeadm feature-gate** for the kubeadm commands

Reply:
For example: https://github.com/kubernetes/kubeadm/blob/2f7c488859fb8c75b4037903dc7895bf05ccec1c/kinder/pkg/cluster/manager/actions/kubeadm-config.go#L97-L100

Reply:
Sorry, It seems to be limited at the beginning.

https://github.com/kubernetes/kubeadm/blob/2f7c488859fb8c75b4037903dc7895bf05ccec1c/kinder/cmd/kinder/do/do.go#L126-L130